sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/RsaKeyCryptographyClient.java

```diff
@@ -132,16 +132,23 @@ private KeyPair getKeyPair(JsonWebKey key) {
 
     @Override
     Mono<SignResult> signAsync(SignatureAlgorithm algorithm, byte[] digest, Context context, JsonWebKey key) {
-
-        return serviceClient.sign(algorithm, digest, context);
+        if (serviceCryptoAvailable()) {
+            return serviceClient.sign(algorithm, digest, context);
+        } else {
+            return Mono.error(new UnsupportedOperationException("Sign operation on Local RSA key"
+                                                                    + " is not supported currently."));
+        }
     }
 
     @Override
     Mono<VerifyResult> verifyAsync(SignatureAlgorithm algorithm, byte[] digest, byte[] signature, Context context,
                                    JsonWebKey key) {
-
-        return serviceClient.verify(algorithm, digest, signature, context);
-        // do a service call for now.
+        if (serviceCryptoAvailable()) {
+            return serviceClient.verify(algorithm, digest, signature, context);
+        } else {
+            return Mono.error(new UnsupportedOperationException("Verify operation on Local RSA key is not"
```

Comment:
Use `monoError()` in `FluxUtil`  which will also log the error. Do the same in other places too.